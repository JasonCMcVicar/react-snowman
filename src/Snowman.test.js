import Snowman from "./Snowman";
import { render, fireEvent } from "@testing-library/react";

//smoke test
it("it renders without crashing", function (){
  render(<Snowman />);
});

it("checking to see that guess options disappear after max guesses", function (){
  const { container } = render(<Snowman words={['cat']} maxWrong={3}/>);

  //Clicks 3 wrong guesses
  fireEvent.click(container.querySelector('button[value="x"]'));
  fireEvent.click(container.querySelector('button[value="y"]'));
  fireEvent.click(container.querySelector('button[value="z"]'));

  const letterButtons = container.querySelector(".letter-buttons");

  expect(letterButtons).toHaveClass('hidden');
  expect(container.querySelector('.result')).toBeInTheDocument();
})
