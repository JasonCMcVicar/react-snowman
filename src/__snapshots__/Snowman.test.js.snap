// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking to see that guess options disappear after max guesses 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="3"
      src="3.png"
    />
    <p
      class="number-wrong"
    >
      Number of wrong guesses: 
      3
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
    </p>
    <p
      class="letter-buttons hidden"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        value="b"
      >
        b
      </button>
      <button
        value="c"
      >
        c
      </button>
      <button
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        value="f"
      >
        f
      </button>
      <button
        value="g"
      >
        g
      </button>
      <button
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        disabled=""
        value="x"
      >
        x
      </button>
      <button
        disabled=""
        value="y"
      >
        y
      </button>
      <button
        disabled=""
        value="z"
      >
        z
      </button>
    </p>
    <div
      class="result"
    >
      You lose!
      <p>
        The word was 
        cat
      </p>
    </div>
  </div>
</div>
`;
